### ISSUE TITLE
Warning about private roads incorrectly displayed for Bicycle (motor_vehicle=private)

### ISSUE THREAD

**Issue by mdykierek** [osmandapp/OsmAnd#17507]
Navigation to locations on this street: https://www.openstreetmap.org/way/109177359#map=15/49.3099/18.1059
shows the following warning:
"Your destination is located in an area with private access. Allow using private roads for this trip?"
This street is perfectly OK for bicycles (explicit Yes) and foot (implicit Yes) access. Actually it is a part of "Cyclostezka Becva" cycle route.
I think it would be better to read access that matches type of vehicle (if possible e.g. foot, bicycle, car, etc.)
It also seems that if the start and destination are at the ends of this road, no warning is shown but the routing avoids the road and creates a long detour despite IT IS THE BEST route.

### Routing engine

- [x] OsmAnd's in-app offline routing
- [ ] Any online routing provider (YOURS, openrouteservice, OSRM, etc.)

### Routing Profile

Bicycle

### Actual and expected routes

No warning

### Is this a regression?

Don't know.

## 🌍 Your Environment

**OsmAnd Version:** 4.4.7

**Device and Android/iOS version:** Mi 9t Pro

**Maps used (online or offline):** Offline

- [x] Offline maps offered within the OsmAnd app for download.
- [ ] Online (tile / raster) maps 

**Anything else relevant?**

**Comment by vshcherb** [comment]
Don't see problems on web version https://osmand.net/map?start=49.309610%2C18.117356&end=49.310883%2C18.099890&mode=bicycle#16/49.3090/18.1075

**Comment by mdykierek** [comment]
It is easily reproducible in the area as a few of these roads are restricted for motor vehicles. Here are my results in the App. Maybe web version is a different piece of routing software:
![1688157191913](https://github.com/osmandapp/OsmAnd/assets/27214784/0e2912a3-5503-40bd-984b-bb011c350413)
![1688157075127](https://github.com/osmandapp/OsmAnd/assets/27214784/900779ce-2d54-46f0-a64c-fbbcb817b334)
As a **bonus** I got several **crashes** when trying to setup the route with GPS off and switching from car profile (the default) to bicycle when trying to select starting point on map.
[exception.log](https://github.com/osmandapp/OsmAnd/files/11923118/exception.log)

**Comment by DmitryAlexei** [comment]
OsmAnd~ 4.5.0#367m, released: 2023-07-10
The same result as in the web version. Private access disabled.

Please check, you may have specified **vehicle routing** in the navigation settings for the bicycle profile.  
https://osmand.net/docs/user/navigation/guidance/navigation-settings#navigation-type

<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/116079541/c58a3d73-a137-4b22-9672-4ec03bc37aa7"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/osmandapp/OsmAnd/assets/116079541/1316a87b-79d2-4e08-8111-f3c74ac87ff9"  width="300" height="auto"/> </th>
  </tr>
</table>

**Comment by mdykierek** [comment]
I had to change my phone and now it seemed harder to reproduce, so the theory with wrong navigation profile is plausible unlikely - I actually managed to generate exact same result as in my previous comment.
![Screenshot_20230715-230139](https://github.com/osmandapp/OsmAnd/assets/27214784/2b57a5fb-4ab0-4986-b877-b7a038d3cf30)
But I nailed it. The routing is "worse" because of elevation data being enabled rather than the private access.

There is definitely is something wrong with the interpretation of "private" access and the dialog notifying about it.
1. Navigation from this way: https://www.openstreetmap.org/way/375793921 to the adjacent way: https://www.openstreetmap.org/way/109177284 causes the "private" access dialog. It is marked private but for motor_vehicle only. The same is true for next segment of cycle route: https://www.openstreetmap.org/way/674286868.
![Screenshot_20230715-224332](https://github.com/osmandapp/OsmAnd/assets/27214784/dc21d9c2-3bbe-4511-8dcf-b712ec135931)

2. The dialog reads "Allow using private roads for this trip?" but it does not seem to care a lot if I answer Yes or No for bicycle.

3. For Android Auto the dialog appears on the phone and waits for input so that the route does not start. Is it possible to display the choices on AA also?

Bottom line is - routing is most probably affected by elevation data not the private access. The "private" road dialog is definitely broken and needs both fixing and improvement (2 and  3).

**Comment by DmitryAlexei** [comment]
**Bicycle routing**: OsmAnd reads tag motor_vehicle=private ([taginfo](https://taginfo.openstreetmap.org/tags/motor_vehicle=private#overview)) and proposes to grant permission to use such a road.
 
It is important to note, that 
1.**destination point should be located on the route tagged with motor_vehicle=private**. If the destination point is after the segment with this tag, the routing is built without warnings. 
2. When using **replace destinationa point** there is no warning.
3. When destination point near the road tagged motor_vehicle=private, may be no warning.

<table style="width:100%">
  <tr>
    <th> 1.  </th>
    <th> 2.  </th>
    <th> 1. ok  </th>
    <th> 2. ok  </th>
  </tr>
  <tr>
    <td><img src="https://github.com/osmandapp/OsmAnd/assets/116079541/b8d0643a-ac67-4430-a100-1b87ef0ef5e4"  width="300" height="auto"/> </td>
    <td><img src="https://github.com/osmandapp/OsmAnd/assets/116079541/4c698af4-3746-42ff-8e27-13456cc08c8d"  width="300" height="auto"/> </td>
    <td><img src="https://github.com/osmandapp/OsmAnd/assets/116079541/7bf9c396-4ef7-4b67-aa5d-8722943c8eb4"  width="300" height="auto"/> </td>
    <td><img src="https://github.com/osmandapp/OsmAnd/assets/116079541/7a925086-f267-403c-9427-6580d0b68ef3"  width="300" height="auto"/> </td>
  </tr>
</table>

1. [OSM](https://www.openstreetmap.org/way/83627392), [OsmAnd Map](https://osmand.net/map/?start=49.303820,18.090247&end=49.305833,18.083278&type=osmand&profile=bicycle#16/49.3055/18.0815)
2. [OSM](https://www.openstreetmap.org/way/109177284), [OsmAnd Map](https://osmand.net/map/?start=49.308043,18.067859&end=49.307931,18.069488&type=osmand&profile=bicycle#16/49.3059/18.0762)

Also tested on https://www.openstreetmap.org/way/51784728 and https://www.openstreetmap.org/way/255036528. Same result, warning appears.

**Comment by dmpr0** [comment]
# Review 
* ToDo is missed

**Comment by vshcherb** [comment]
Expected: - https://www.openstreetmap.org/way/109177359#map=15/49.3099/18.1059 (used motorvehicle "private road")
https://osmand.net/map/?start=49.311946,18.131990&finish=49.310883,18.099890&type=osmand&profile=bicycle#15/49.3126/18.1064
![Screenshot 2023-11-21 at 15 55 41](https://github.com/osmandapp/OsmAnd/assets/1042025/262443f5-d770-44d6-9413-8dc8ee7758e9)

Before: https://www.openstreetmap.org/way/109177359#map=15/49.3099/18.1059 road wasn't used and route was longer
